department-of-veterans-affairs/abd-vro
# BIE: Update svc-bie-kafka to persist contention event in the application

## What was the problem?

Current persistence of BIE Contention Events involves multiple services: `svc-bie-kafka`, RabbitMQ, and `xample-workflow`. This could potentially result in multiple points of failure just to persist the contention events in RDS. 

Associated tickets or Slack threads:
- #3334 
- #3329 

## How does this fix it?[^1]

* removes the persistence from the `xample-workflow` application
* moves it into `svc-bie-kafka` 
* updates rabbitmq declarations to match config set in [#3329 ](https://github.com/department-of-veterans-affairs/abd-vro/issues/3329)
* updates unit/integration tests

### Note
Another PR will follow to remove the `xample-workflow` altogether.

## How to test this PR
1. Disable SSL by following comment on line 68 in `mocks/docker-compose.yml` and changing `svc-bie-kafka` application properties file to have 
2. Build and run platform 
```
source scripts/setenv.sh
./gradlew clean docker
COMPOSE_PROFILES="platform" ./gradlew :dockerComposeUp
```
3. Build and run mock 
```
./gradlew -p mocks docker 
COMPOSE_PROFILES="kafka" ./gradlew -p mocks :dockerComposeUp
```
4. Build and run `svc-bie-kafka`
```
COMPOSE_PROFILES="platform,kafka" ./gradlew :dockerComposeUp
```
5. Start kafka console producer on the topic:
```
kafka-console-producer.sh --bootstrap-server localhost:9092 --topic BIA_SERVICES_BIE_CATALOG_TST_CONTENTION_CLASSIFIED_V02
```

> _Note: `kafka-console-producer.sh` is included in the `vro-mock-bie-kafka` bitnami image, and will be callable in the container console._

5. Send mock kafka message after the `<` indicates the producer is listening:
```
{"claimId":-3202217447607305195,"contentionId":-2729257799809135899,"diagnosticTypeCode":"a3Z251Ha5F","contentionTypeCode":"h5Vy092Rbt","contentionClassificationName":"blanditiis","actionName":"33948q0Et4","actionResultName":"406j54W9H4","dateAdded":null,"dateCompleted":null,"dateUpdated":null,"actorStation":"55345ck592483K239Edxy8618N9953Aly280Qr06764Cwemo449P3Rnf5923Ap431R4D83Wgz4Aclwni6L384325Wiqy2B77Fx40V3B381D863Mjn83369U90R3D86Sv6Y06Qk359Avqe397H2Pt4D92L6S16Lv093Eo13520F4Aeyz0J1I2Au2Zn948Ng191Op29Ry4","automationIndicator":false,"benefitClaimTypeCode":"l98A0Xk971","contentionStatusTypeCode":"811bd","currentLifecycleStatus":"6w870Xuq","description":null,"details":"3i5R630S9A644Ysm231570P69B2Qca0149382Dm1Bucg4I3Tt2Ouecit7G43Ura39M8Wsseo3Zoscv6Hso75Qv93R2G23Jhgh6Flzo726Cx50991R9Vym4Oiban51Ivw7Ve4Ube394J2P2Waq58Qx3Zz2Am3Z0Dni08O9Rf3Xmd4X2Mprgtysxxuv3N53X9Byao3Urdf","journalStatusTypeCode":null,"veteranParticipantId":"-8152877461276601739","eventTime":1725646029,"actorUserId":null}
```
6. Verify data was saved in postgres via Postico or other tool

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
